File: Data Structures & Algorithms/anagram-groups/submission-3.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mp;
        vector<vector<string>> res;
        for(auto s : strs){
            unordered_map<int, int> count;
            for(auto c : s){
                count[c-'a']++;
            }
            string key;
            for(int c = 0; c < 26; c++){
                key += "," + to_string(count[c]);
            }
            mp[key].push_back(s);
        }
        for(auto i : mp){
            res.push_back(i.second);
        }
        return res;
    }
};

File: Data Structures & Algorithms/valid-palindrome-ii/submission-3.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        l, r = 0, len(s)-1
        count = 0
        def ispali(l, r):
            while l < r:
                if s[l] != s[r]:
                    return False
                l+=1
                r-=1
            return True
        while l < r:
            if s[l] != s[r]:
                return ispali(l+1, r) or ispali(l, r-1)
            l+=1
            r-=1
        return True
